lib/rules/declaration-property-value-keyword-no-deprecated/index.mjs

```diff
@@ -129,10 +129,17 @@ const rule = (primary, secondaryOptions) => {
 			if (!keywords) return;
 
 			const parsedValue = valueParser(getDeclarationValue(decl));
+			let flag = false;
 
 			parsedValue.walk((node) => {
+				if (flag) return;
```

Comment:
@romainmenke I tried returning `false` + setting bubble to `true`; how do you break from `walk`?

Reply:
> Can you add failing tests?

I can't.
see https://github.com/stylelint/stylelint/issues/8327#issuecomment-2596623482

But you can try `npm i -D postcss-styled-syntax` +
```js
testRule({
	ruleName,
	customSyntax: 'postcss-styled-syntax',
	config: [true],

	accept: [
		{
			code: `
				export const s = styled.a\`
					background-color: \${palette(({ background }) => background.primary)};
				\`;
			`,
		},
	],
});
```

Reply:
Shouldn't we just return early when the value isn't a standard syntax value?
Currently we only check the property, not the value.

Reply:
I think that not running the code unnecessarily is still a good thing.
It will also skip cases that are invalid.

e.g.
someone who erroneously used `background-color` instead of `background-color`
`background-color: border-box WindowFrame;`

---

> Ill try to come up with a non css-in-js test.

I can't come up with a non-invalid CSS test case because we don't support shorthands yet.
i.e. the properties are filtered
see also #8328

Reply:
If there is no failing test or real world use case then I can't really advise :/

Reply:
If the concern is purely performance and that this change isn't intended to change the behavior then I would suggest adding an early check that the value at least contains one of the candidates. see: https://github.com/stylelint/stylelint/issues/6869

Skipping the parsing step will be much more impactful than stopping the walk early.

If however there is an intended behavior change, then there should be some illustration of why the current behavior isn't ideal :)
I am still trying to understand what the issue is and how this change relates to it.
Given that the title is also `TBD` I really have too little information to be helpful.
My advice is likely wrong purely because I don't have sufficient information.

Reply:
> Shouldn't we just return early when the value isn't a standard syntax value?

Ill just do that.